### PR TITLE
Reload config

### PR DESCRIPTION
**Pull request by vide**
This PR implements separate reloading/restarting of the prometheus daemon according to the type of configuration updated.
If it's in prometheus' YAML config files (i.e. prometheus.yaml ) then, the daemon just receives an HUP signal (through its native init system) to reload the conf.
If, OTOH, the change is to the command-line parameters, a full restart is issued, cause otherwise a reload won't propagate those changes.

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -31,9 +42,10 @@
           mode    => '0644',
           owner   => 'root',
           group   => 'root',
+          notify  => $systemd_notify,
           content => template('prometheus/prometheus.systemd.erb'),
         }
-        ~> exec { 'prometheus-systemd-reload':
+        exec { 'prometheus-systemd-reload':
```
Comment (bastelfreak) [inline manifests/config.pp]: could you replace this exec with a notify to the one provided here: https://github.com/camptocamp/puppet-systemd

and add the module as dependency to the metadata.json

**Comment by vide** [comment]
@bastelfreak is it ok now?

**Comment by juliantaylor** [comment]
this looks very useful. One thing that seems to be missing is to also trigger a reload when the alert files (alert.rules) changes. Prometheus can also reload these.

**Comment by vide** [comment]
Thx @juliantaylor Still waiting for approval from voxpupuli folks :(

**Comment by bastelfreak** [comment]
Hi @vide, sorry for the slow feedback :(

**Comment by salekseev** [comment]
+1

**Pull request merged**